### PR TITLE
nirbhay/aws_waf_ip_set

### PR DESCRIPTION
**Pull request by Nirbhay1997**
Signed-off-by: Nirbhay Kumar <nkumar@progress.com>

### Description

It contains all the singular and plural resources of the above resource. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_waf_ip_set.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AWSWAFIPSet < AwsResourceBase
+  name 'aws_waf_ip_set'
+  desc 'Describes one WAF IP set.'
+
+  example "
+    describe aws_waf_ip_set(ip_set_id: 'IP_SET_NAME') do
```
Comment (soumyo13) [inline libraries/aws_waf_ip_set.rb:10]:
```suggestion
    describe aws_waf_ip_set(ip_set_id: 'IP_SET_ID') do
```

File: libraries/aws_waf_ip_set.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return nil unless exists?
+    @resp[:ip_set_id]
+  end
+
+  def exists?
+    !@resp.nil? && !@resp.empty?
+  end
+
+  def to_s
+    "Resource ID: #{@display_name}"
```
Comment (soumyo13) [inline libraries/aws_waf_ip_set.rb:38]:
```suggestion
    "IP Set ID: #{@display_name}"
```

File: test/unit/resources/aws_waf_ip_set_test.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  def test_rule_exists
+    assert @resp.exist?
+  end
+
+  def ip_set_id
+    assert_equal(@resp.ip_set_id, 'test1')
+  end
+
+  def test_byte_match_set_name
```
Comment (soumyo13) [inline test/unit/resources/aws_waf_ip_set_test.rb:42]: Change it!

**Review by soumyo13 — CHANGES_REQUESTED** [review]

File: libraries/aws_waf_ip_set.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
```
Comment (soumyo13) [inline libraries/aws_waf_ip_set.rb]:
```suggestion
    "IP Set ID: #{@display_name}"
```

File: docs/resources/aws_waf_ip_set.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+| ip_set_descriptors | The IP address type (IPV4 or IPV6 ) and the IP address range (in CIDR notation) that web requests originate from. | ip_set_descriptors |
+| ip_set_descriptors_types | Specify IPV4 or IPV6. |  ip_set_descriptors (type) |
+| ip_set_descriptors_values | Specify an IPv4 address by using CIDR notation. |  ip_set_descriptors (value) |
+
+## Examples
+
+### Ensure a ip set is available.
+
+    describe aws_waf_ip_set(ip_set_id: 'IP_SET_ID') do
+      its('ip_set_id') { should eq 'WEB_ACL_ID' }
```
Comment (soumyo13) [inline docs/resources/aws_waf_ip_set.md:41]:
```suggestion
      its('ip_set_id') { should eq 'IP_SET_ID' }
```

File: docs/resources/aws_waf_ip_set.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    end
+
+### Ensure a ip set name is available..
+
+    describe aws_waf_ip_set(ip_set_id: 'IP_SET_ID') do
+        its('name') { should eq 'WEB_ACL_NAME' }
```
Comment (soumyo13) [inline docs/resources/aws_waf_ip_set.md:47]:
```suggestion
        its('name') { should eq 'IP_SET_NAME' }
```

File: docs/resources/aws_waf_ip_sets.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+| --- | --- | --- |
+| ip_set_ids | The IPSetId for an IPSet. | ip_set_id |
+| names | A friendly name or description of the IPSet. | name |
+
+## Examples
+
+### Ensure a ip set is available.
+
+    describe aws_waf_ip_sets do
+      its('ip_set_id') { should include 'WEB_ACL_ID' }
```
Comment (soumyo13) [inline docs/resources/aws_waf_ip_sets.md:38]:
```suggestion
      its('ip_set_ids') { should include 'IP_SET_ID' }
```

File: docs/resources/aws_waf_ip_sets.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+### Ensure a ip set name is available..
+
+    describe aws_waf_ip_sets do
+        its('name') { should include 'WEB_ACL_NAME' }
```
Comment (soumyo13) [inline docs/resources/aws_waf_ip_sets.md:44]:
```suggestion
        its('names') { should include 'IP_SET_NAME' }
```

File: test/unit/resources/aws_waf_ip_set_test.rb (hunk 3 above)
Comment (Nirbhay1997) [inline test/unit/resources/aws_waf_ip_set_test.rb:42]:
```suggestion
  def test_ip_set_name
```

File: test/unit/resources/aws_waf_ip_set_test.rb [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    mock_data = {}
+    mock_data[:ip_set_id] = 'test1'
+    mock_data[:name] = 'test1'
+    mock_data[:ip_set_descriptors] = [{ type: "test1", value: 'test1' }]
+    data[:data] = { ip_set: mock_data }
+    data[:client] = Aws::WAF::Client
+    @resp = AWSWAFIPSet.new(ip_set_id: 'test1', client_args: { stub_responses: true }, stub_data: [data])
+  end
+
+  def test_rule_exists
```
Comment (Nirbhay1997) [inline test/unit/resources/aws_waf_ip_set_test.rb:34]:
```suggestion
  def test_ip_set_exists
```

File: test/unit/resources/aws_waf_ip_sets_test.rb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    data = {}
+    data[:method] = :list_ip_sets
+    mock_data = {}
+    mock_data[:ip_sets] = [{ ip_set_id: 'ip_set_id', name: 'name' }]
+    data[:data] = mock_data
+    data[:client] = Aws::WAF::Client
+    @resp = AWSWAFIPSets.new(client_args: { stub_responses: true }, stub_data: [data])
+  end
+
+  def test_rule_exists
```
Comment (Nirbhay1997) [inline test/unit/resources/aws_waf_ip_sets_test.rb:28]:
```suggestion
  def test_ip_set_exists
```

File: test/unit/resources/aws_waf_ip_sets_test.rb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    assert @resp.exist?
+  end
+
+  def test_rule_ids
```
Comment (Nirbhay1997) [inline test/unit/resources/aws_waf_ip_sets_test.rb:32]:
```suggestion
  def test_ip_set_ids
```

File: test/unit/resources/aws_waf_ip_set_test.rb (hunk 9 above)
Comment (Nirbhay1997) [inline test/unit/resources/aws_waf_ip_set_test.rb]:
```suggestion
  def test_ip_set_exists
```

Comment (Nirbhay1997) [inline test/unit/resources/aws_waf_ip_set_test.rb]:
```suggestion
  def test_ip_set_exists
```

**Review by soumyo13 — APPROVED** [review]

**Review by IanMadd — APPROVED** [review]

**Pull request merged**